### PR TITLE
KAFKA-6179: Clear min timestamp tracker upon partition queue cleanup

### PR DESCRIPTION
**Pull request by guozhangwang**

### CODE REVIEW

**Comment by guozhangwang** [comment]
Will try to add a unit test that fails before this patch to validate.

**Comment by asfgit** [comment]
FAILURE
 No test results found.
--none--

**Comment by asfgit** [comment]
SUCCESS 
 8053 tests run, 5 skipped, 0 failed.
--none--

**Comment by asfgit** [comment]
FAILURE
 7945 tests run, 5 skipped, 1 failed.
--none--

File: streams/src/main/java/org/apache/kafka/streams/processor/internals/MinTimestampTracker.java [hunk 1]
```diff
@@ -77,4 +77,8 @@ public long get() {
             return stamped.timestamp;
     }
 
+    public void clear() {
```
Comment (mjsax) [inline streams/src/main/java/org/apache/kafka/streams/processor/internals/MinTimestampTracker.java:80]: Where do we call this?

Comment (guozhangwang) [reply streams/src/main/java/org/apache/kafka/streams/processor/internals/MinTimestampTracker.java:80]: It is called in `RecordQueue.clear()`.

**Comment by asfgit** [comment]
FAILURE
 No test results found.
--none--

**Comment by bbejeck** [comment]
Just one comment, maybe we should update the javadoc on `RecordQueue#clear()` to include the `TimestampTracker` is cleared as well.  Otherwise LGTM.

**Comment by asfgit** [comment]
FAILURE
 7945 tests run, 5 skipped, 1 failed.
--none--

**Comment by asfgit** [comment]
SUCCESS 
 8053 tests run, 5 skipped, 0 failed.
--none--

**Comment by asfgit** [comment]
FAILURE
 7267 tests run, 1 skipped, 0 failed.
--none--

**Review by dguy — APPROVED** [review]
LGTM

**Comment by guozhangwang** [comment]
> Just one comment, maybe we should update the javadoc on RecordQueue#clear() to include the TimestampTracker is cleared as well

Ack, will do that.

File: streams/src/main/java/org/apache/kafka/streams/processor/internals/MinTimestampTracker.java (hunk 1 above)
Comment (mjsax) [reply streams/src/main/java/org/apache/kafka/streams/processor/internals/MinTimestampTracker.java:80]: Now it is :) You added some more commits :)

**Review by mjsax — APPROVED** [review]
LGTM.

File: streams/src/main/java/org/apache/kafka/streams/processor/internals/MinTimestampTracker.java (hunk 1 above)
Comment (guozhangwang) [reply streams/src/main/java/org/apache/kafka/streams/processor/internals/MinTimestampTracker.java:80]: That's right. I missed adding the diff when committing for the first time.

**Comment by guozhangwang** [comment]
Merged to trunk, 1.0 and 0.11.0 (cc @rajinisivaram ).

**Pull request closed without merging**